docs/_pages/contributing.md

```diff
@@ -130,6 +126,65 @@ and more importantly also try to improve the process along the way as well.
    - Before your change can be merged, it should be squashed into a single commit for cleaner commit history.
  - Finally, once your pull request is merged, make sure to `Close` the JIRA.
 
+### Coding guidelines 
+
+Our code can benefit from contributors speaking the same "language" when authoring code. After all, it gets read lot more than it gets
+written. So optimizing for "reads" is a good goal. The list below is a set of guidelines, that contributors strive to upkeep and reflective 
+of how we want to evolve our code in the future.
+
+#### Style 
+
+ - **Formatting** We should rely on checkstyle and spotless to auto fix formatting; automate this completely. Where we cannot,
+    we will err on the side of not taxing contributors with manual effort.
+ - **Refactoring**
+   - Refactor with purpose; any refactor suggested should be attributable to functionality that now becomes easy to implement.
+   - A class is asking to be refactored, when it has several overloaded responsibilities/have sets of fields/methods which are used more cohesively than others. 
+   - Try to name tests using the given-when-then model, that cleans separates preconditions (given), an action (when) and assertions (then).
+ - **Naming things**
+   - Let's name uniformly; using the same word to denote the same concept. e.g: bootstrap vs external vs source, when referring to bootstrapped tables. 
+     May be they all mean the same, but having one word makes the code lot more easily readable. 
+   - Let's name consistently with Hudi terminology. e.g dataset vs table, base file vs data file..
+   - Class names preferably are nouns (e.g Runner) which reflect their responsibility and methods are verbs (e.g run()).
+   - Avoid filler words, that don't add value e.g xxxInfo, xxxData etc.
+   - We name classes in code starting with `Hoodie` and not `Hudi` and we want to keep it that way for consistency/historical reasons. 
+ - **Methods**
+   - Individual methods should short (~20-30 lines) and have a single purpose; If you feel like it has a secondary purpose, then may be it needs
+     to be broken down more.
+   - Lesser the number of arguments, the better; 
+   - Place caller methods on top of callee methods, whenever possible.
```

Comment:
do we have any guidelines for ordering or public, package private and private method. Also, in tests, actual test methods and private/utility methods and assertion methods.

Reply:
this would go counter to that sort of ordering. most of this is distilled from the clean code book, which recomments logical grouping by cohesion, rather than artificially by private/public